cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -0,0 +1,380 @@
+
+        # Load up the Certificate data
+        certificate_data = CertificateManager.parse(value)
+        CertificateManager.validate(certificate_data)
+        certificate_data = CertificateManager.assign_id(course, certificate_data, certificate_data.get('id', None))
+        certificate = Certificate(course, certificate_data)
+
+        # Return a new Certificate object instance
+        return certificate
+
+    @staticmethod
+    def deserialize_certificate(course, json_string):
+        """
+        Returns a Certificate object instance given a valid course and JSON representation
+        """
+        return CertificateManager.from_json(course, json_string)
+
+    @staticmethod
+    def get_certificates(course):
+        """
+        Retrieve the certificates list from the provided course
+        """
+        # The top-level course field is 'certificates'
+        course_certificates_field = getattr(course, 'certificates')
+        # Inside of this top-level field are various things, including the actual 'certificates' list
+        certificates = course_certificates_field.get('certificates', [])
+        return certificates
+
+    @staticmethod
+    def remove_certificate(request, store, course, certificate_id):
+        """
+        Remove certificate from the course
+        """
+        for index, cert in enumerate(course.certificates['certificates']):
+            if int(cert['id']) == int(certificate_id):
+                course.certificates['certificates'].pop(index)
+                store.update_item(course, request.user.id)
+                break
+        return JsonResponse(status=204)
+
+    # pylint-disable: unused-variable
+    @staticmethod
+    def remove_signatory(request, store, course, certificate_id, signatory_id):
+        """
+        Remove the specified signatory from the provided course certificate
+        """
+        for cert_index, cert in enumerate(course.certificates['certificates']):  # pylint: disable=unused-variable
+            if int(cert['id']) == int(certificate_id):
+                for sig_index, signatory in enumerate(cert.get('signatories')):  # pylint: disable=unused-variable
+                    if int(signatory_id) == int(signatory['id']):
+                        del cert['signatories'][sig_index]
+                        store.update_item(course, request.user.id)
+                        break
+        return JsonResponse(status=204)
+
+
+class Certificate(object):
+    """
+    The logical representation of an individual course certificate
+    """
+    def __init__(self, course, certificate_data):
+        """
+        Instantiate a Certificate object instance using the provided information.
+        """
+        self.course = course
+        self.certificate_data = certificate_data
+        self.id = certificate_data['id']  # pylint: disable=invalid-name
+
+    def get_certificate_data(self):
```

Comment:
Could make this a @property, then I think one can remove the `# pylint: disable=no-member` flag when calling above... although I'm not even sure that it's needed anyway (I don't appear to get the error using a local pylint plugin)